stanford_profile_helper.module

```diff
@@ -260,3 +262,124 @@ function stanford_profile_helper_config_readonly_whitelist_patterns() {
+    ],
+  ];
+
+  // LINE 1.
+  $form['su_line_1']['widget']['0']['value']['#states'] = [
+    'invisible' => _stanford_profile_helper_get_lockup_states(
+      ['_none', 'none', 'o', 'r'],
+      ':input[name="su_lockup_options"]'
+    ),
+  ];
+
+  // LINE 2.
+  $form['su_line_2']['widget']['0']['value']['#states'] = [
+    'invisible' => _stanford_profile_helper_get_lockup_states(
+      ['_none', 'none', 'a', 'd', 'h', 'i', 'k', 'l', 'n', 'o', 'p', 'q', 'r'],
+      ':input[name="su_lockup_options"]'
+    ),
+  ];
+
+  // LINE 3.
+  $form['su_line_3']['widget']['0']['value']['#states'] = [
+    'invisible' => _stanford_profile_helper_get_lockup_states(
+      ['_none', 'none', 'a', 'b', 'c', 'f', 'g', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's'],
+      ':input[name="su_lockup_options"]'
+    ),
+  ];
+
+  // LINE 4.
+  $form['su_line_4']['widget']['0']['value']['#states'] = [
+    'invisible' => _stanford_profile_helper_get_lockup_states(
+      ['_none', 'none', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'j', 'k', 'l', 'm', 'n', 'r'],
+      ':input[name="su_lockup_options"]'
+    ),
+  ];
+
+  // LINE 5.
+  $form['su_line_5']['widget']['0']['value']['#states'] = [
+    'invisible' => _stanford_profile_helper_get_lockup_states(
+      ['_none', 'none', 'b', 'd', 'e', 'f', 'h', 'i', 'l', 'm', 'n', 'o', 'p', 'q', 's', 't'],
+      ':input[name="su_lockup_options"]'
+    ),
+  ];
+
+  // Clear caches on submit.
+  $form['actions']["submit"]['#submit'][] = "stanford_profile_helper_form_config_pages_lockup_settings_form_alter_submit";
+}
+
+/**
+ * Implements field_group_form_process_build_alter().
+ */
+function stanford_profile_helper_field_group_form_process_build_alter(&$element) {
+  // Hide / Show the field groups based on the enabled checkbox.
+  if (isset($element['group_lockup_options'])) {
+    $element['group_lockup_options']['#states'] = [
+      'visible' => [
+        ':input[name="su_lockup_enabled[value]"]' => [
+          'checked' => FALSE,
+        ],
+      ],
+    ];
+    $element['group_logo_image']['#states'] = [
+      'visible' => [
+        ':input[name="su_lockup_enabled[value]"]' => [
+          'checked' => FALSE,
+        ],
+      ],
+    ];
+  }
+}
+
+/**
+ * Creates a states array.
+ *
+ * @param array $opts
+ *   Allowed values.
+ * @param string $input
+ *   Field selector.
+ *
+ * @return array
+ *   State array.
+ */
+function _stanford_profile_helper_get_lockup_states(array $opts, $input) {
+  $ret = [];
+  foreach ($opts as $val) {
+    $ret[] = [
+      $input => ['value' => $val],
+    ];
+  }
+  return $ret;
+}
+
+/**
+ * Clear cache after lockup config page form submit.
+ *
+ * @param array $form
+ *   The form array.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The form state.
+ */
+function stanford_profile_helper_form_config_pages_lockup_settings_form_alter_submit(array &$form, FormStateInterface $form_state) {
+  drupal_flush_all_caches();
```

Comment:
oof. thats a heavy cache clear. can we just clear render caches?

Reply:
I tried and varnish didn't seem to pick it up.

Reply:
These changes should be one and done so one big cache clear doesn't scare me.

Reply:
_should be_ will be written on your tomb stone. :rofl:

Reply:
we can improve it later by adding a cache tag to every page

Reply:
Aye, I hope it doesn't get abused.

Reply:
ooh we could just clear the tag `config:system.site` that should flush render and varnish.

Reply:
Have at it.

Reply:
I shall.

Reply:
```suggestion
  Cache::invalidateTags(['config:system.site']);
```
@sherakama that should do it.